geoengine/types.py

```diff
@@ -656,3 +697,93 @@ def measurement(self) -> str:
     @property
     def classes(self) -> Dict[int, str]:
         return self.__classes
+
+
+@dataclass
+class GdalDatasetGeoTransform:
+    '''Geo transform of a GDAL dataset'''
+    origin_coordinate: Tuple[float, float]
+    x_pixel_size: float
+    y_pixel_size: float
+
+    def to_dict(self) -> Dict[str, Any]:
+        return {
+            "originCoordinate": {
+                "x": self.origin_coordinate[0],
+                "y": self.origin_coordinate[1],
+            },
+            "xPixelSize": self.x_pixel_size,
+            "yPixelSize": self.y_pixel_size,
+        }
+
+
+class FileNotFoundHandling(Enum):
+    NODATA = "NoData"
+    ERROR = "Abort"
+
+
+@dataclass
+class RasterPropertiesKey:
+    '''Key of a raster properties entry'''
+    domain: Optional[str]
+    key: str
+
+
+class RasterPropertiesEntryType(Enum):
+    NUMBER = "number"
+    STRING = "string"
+
+
+@dataclass
+class GdalMetadataMapping:
+    '''Mapping of GDAL metadata raster properties'''
+
+    source_key: RasterPropertiesKey
+    target_key: RasterPropertiesKey
+    target_type: RasterPropertiesEntryType
+
+    def to_dict(self) -> Dict[str, Any]:
+        return {
+            "sourceKey": {
+                "domain": self.source_key.domain,
+                "key": self.source_key.key,
+            },
+            "targetKey": {
+                "domain": self.target_key.domain,
+                "key": self.target_key.key,
+            },
+            "targetType": self.target_type.value,
+        }
+
+
+@dataclass
+class GdalDatasetParameters:
+    '''Parameters for a GDAL dataset'''
+
+    file_path: str
+    rasterband_channel: int
+    geo_transform: GdalDatasetGeoTransform
+    width: int
+    height: int
+    file_not_found_handling: FileNotFoundHandling
+    no_data_value: Optional[float]
+    properties_mapping: Optional[List[GdalMetadataMapping]]
+    gdal_open_options: Optional[List[str]]
+    gdal_config_options: Optional[List[Tuple[str, str]]]
+    allow_alphaband_as_mask: bool
+
+    def to_dict(self) -> Dict[str, Any]:
```

Comment:
and here

Reply:
04e3b2b514c10eeaa30770a65b6b2d4663500498